src/gt4py/next/program_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -326,12 +330,60 @@ def _collect_offset_or_axis_node(
             )
         return result
 
-    def _error_on_illegal_function_calls(self, node: itir.FunCall) -> None:
-        assert isinstance(node.fun, itir.SymRef)
-        if node.fun.id == "shift":
-            raise ValueError("unapplied shift call not supported: {node}")
-        elif node.fun.id == "scan":
-            raise ValueError("scans are only supported at the top level of a stencil closure")
+    def _visit_if_(self, node: itir.FunCall, **kwargs: Any) -> Node:
+        assert len(node.args) == 3
+        return TernaryExpr(
+            cond=self.visit(node.args[0], **kwargs),
+            true_expr=self.visit(node.args[1], **kwargs),
+            false_expr=self.visit(node.args[2], **kwargs),
+        )
+
+    def _visit_cast_(self, node: itir.FunCall, **kwargs: Any) -> Node:
+        assert len(node.args) == 2
+        return CastExpr(
+            obj_expr=self.visit(node.args[0], **kwargs),
+            new_dtype=self.visit(node.args[1], **kwargs),
+        )
+
+    def _visit_tuple_get(self, node: itir.FunCall, **kwargs: Any) -> Node:
+        assert isinstance(node.args[0], itir.Literal)
+        return FunCall(
+            fun=SymRef(id="tuple_get"),
+            args=[
+                self.visit(node.args[0], as_integral_constant=True),
+                self.visit(node.args[1]),
+            ],
+        )
+
+    def _visit_list_get(self, node: itir.FunCall, **kwargs: Any) -> Node:
+        # should only reach this for the case of an external sparse field
+        return FunCall(
+            fun=SymRef(id="tuple_get"),
+            args=[
+                self.visit(node.args[0], as_integral_constant=True),
```

Comment:
what I had in mind was to make this work also for expressions of integral_constants, but doesn't make sense at this point, because:
- there is an assert that node.args[0] is a literal
- not all builtins work on integral_constants

fixed as suggested